src/Button/RedliningButton/RedliningButton.jsx

```diff
@@ -0,0 +1,672 @@
+   * Checks if a labeled feature is being modified. If yes, opens prompt to
+   * input a new label.
+   *
+   * @param {Event} evt 'modifystart' event of OlInteractionModify.
+   *
+   * @method
+   */
+  onModifyStart = evt => {
+    const feature = evt.features.getArray()[0];
+    if (feature.get('isLabel')) {
+      this._redliningTextFeature = feature;
+      this.setState({
+        showLabelPrompt: true
+      });
+    }
+  }
+
+  /**
+   * Changes state for showLabelPrompt to make modal for label input visible.
+   *
+   * @param {Event} evt Click event on adding feature to the redlining layer.
+   *
+   * @method
+   */
+  handleTextAdding = evt => {
+    this.setState({
+      showLabelPrompt: true
+    });
+    this._redliningTextFeature = evt.element;
+    this._redliningTextFeature.set('isLabel', true);
+  }
+
+  /**
+   * Callback function after `Ok` button of label input modal was clicked.
+   * Turns visibility of modal off and call `setTextOnFeature` method.
+   *
+   * @method
+   */
+  onModalLabelOk = () => {
+    this.setState({
+      showLabelPrompt: false
+    }, () => {
+      this.setTextOnFeature(this._redliningTextFeature);
+    });
+  }
+
+  /**
+   * Callback function after `Cancel` button of label input modal was clicked.
+   * Turns visibility of modal off and removes last drawn feature from the
+   * redlining layer.
+   *
+   * @method
+   */
+  onModalLabelCancel = () => {
+    this.setState({
+      showLabelPrompt: false,
+      textLabel: ''
+    }, () => {
+      if (!(this.state.interaction && this.state.interaction.length > 1)) {
+        this._redliningFeatures.remove(this._redliningTextFeature);
+        this._redliningTextFeature = null;
+      }
+    });
+  }
+
+  /**
+   * Sets formatted label on feature.
+   *
+   * @param {OlFeature} feat The point feature to be styled with label.
+   *
+   * @method
+   */
+  setTextOnFeature = (feat) => {
+    const label = StringUtil.stringDivider(this.state.textLabel, 16, '\n');
+    const style = this.getRedliningStyleFunction(feat).clone();
+    style.getText().setText(label);
+    feat.setStyle(style);
+
+    this.setState({
+      textLabel: ''
+    });
+
+  }
+
+  /**
+   * Called if label input field value was changed. Updates state value for
+   * textLabel.
+   *
+   * @param {Event} evt Input event containing new text value to be set as
+   * textLabel.
+   *
+   * @method
+   */
+  onLabelChange = (evt) => {
+    this.setState({
+      textLabel: evt.target.value
+    });
+  }
+
+  /**
+   * Sets the cursor to `pointer` if the pointer enters a non-oqaque pixel of
+   * a hoverable layer.
+   *
+   * @param {ol.MapEvent} evt The `pointermove` event.
+   *
+   * @method
+   */
+  onPointerMove = evt => {
+    if (evt.dragging) {
+      return;
+    }
+
+    const { map } = this.props;
+    const pixel = map.getEventPixel(evt.originalEvent);
+    const hit = map.hasFeatureAtPixel(pixel);
+
+    map.getTargetElement().style.cursor = hit ? 'pointer' : '';
```

Comment:
Unsure here, but we may want to include a (customizable) `layerFilter` here to only change the cursor for features/layers available to edit with the given interactions.